app/services/file_attachment_blob_service.rb

```diff
@@ -1,41 +1,44 @@
 # frozen_string_literal: true
 
-class FileAttachmentBlobService
-  def initialize(revision, user)
+class FileAttachmentBlobService < ApplicationService
+  def initialize(revision, user, file, replacing: nil)
     @revision = revision
     @user = user
+    @file = file
+    @replacing = replacing
   end
 
-  def create_blob_revision(file, replacing: nil)
-    mime_type = Marcel::MimeType.for(file,
-                                     declared_type: file.content_type,
-                                     name: file.original_filename)
-
-    filename = unique_filename(file, replacing)
+  def call
     blob = ActiveStorage::Blob.create_after_upload!(io: file,
-                                                    filename: filename,
+                                                    filename: unique_filename,
                                                     content_type: mime_type)
 
     FileAttachment::BlobRevision.create!(
       blob: blob,
-      filename: filename,
-      number_of_pages: number_of_pages(file, mime_type),
+      filename: unique_filename,
+      number_of_pages: number_of_pages,
       created_by: user,
       asset: FileAttachment::Asset.new,
     )
   end
 
 private
 
-  attr_reader :revision, :user
+  attr_reader :revision, :user, :file, :replacing
```

Comment:
It seems to be a more common convention in our services classes to have these public. We should probably go one way or the other consistently. Personally I'd vote private.

Reply:
Although the further down I go in this review it might be private that is more common.